messaging/src/main/java/org/axonframework/configuration/Components.java

```diff
@@ -111,4 +112,33 @@ public <C> Component<C> computeIfAbsent(
     public boolean contains(Identifier<?> identifier) {
         return components.containsKey(identifier);
     }
+
+    /**
+     * Returns the identifiers of the components currently registered.
+     *
+     * @return a set with the identifiers of registered components
+     */
+    public Set<Identifier<?>> listComponents() {
+        return Set.copyOf(components.keySet());
+    }
+
+    /**
+     * Replace the component registered under the given {@code identifier} with the instance returned by given
+     * {@code replacement} function. If no component is registered under the given identifier, nothing happens.
+     * <p>
+     * If the given {@code replacement} function returns null, the component registration is removed.
+     *
+     * @param identifier  The identifier of the component to replace
+     * @param replacement The function providing the replacement value, based on the currently registered component
+     * @param <C>         The type of component registered
+     * @return {@code true} if a component is present and has been replaced, {@code false} if no component was present,
+     * or has been removed by the replacement function
+     */
+    public <C> boolean replace(Identifier<C> identifier,
```

Comment:
Nit: There's a `ComponentsTest` class. I think it would be nice to add tests for these new methods.